seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/TextReadStrategy.java

```diff
@@ -94,32 +96,21 @@ public void read(String path, String tableId, Collector<SeaTunnelRow> output)
                     .forEach(
                             line -> {
                                 try {
-                                    SeaTunnelRow seaTunnelRow =
-                                            deserializationSchema.deserialize(line.getBytes());
-                                    if (!readColumns.isEmpty()) {
-                                        // need column projection
-                                        Object[] fields;
-                                        if (isMergePartition) {
-                                            fields =
-                                                    new Object
-                                                            [readColumns.size()
-                                                                    + partitionsMap.size()];
-                                        } else {
-                                            fields = new Object[readColumns.size()];
-                                        }
-                                        for (int i = 0; i < indexes.length; i++) {
-                                            fields[i] = seaTunnelRow.getField(indexes[i]);
-                                        }
-                                        seaTunnelRow = new SeaTunnelRow(fields);
-                                    }
-                                    if (isMergePartition) {
-                                        int index = seaTunnelRowType.getTotalFields();
-                                        for (String value : partitionsMap.values()) {
-                                            seaTunnelRow.setField(index++, value);
+                                    if (StringUtils.isBlank(rowDelimiter)) {
+                                        SeaTunnelRow seaTunnelRow =
+                                                deserializeRow(line, partitionsMap, tableId);
+                                        output.collect(seaTunnelRow);
+                                    } else {
+                                        String[] rows = line.split(rowDelimiter, -1);
```

Comment:
You cannot deal with rowDelimiter at this way, this can not work in most of case, since not all data will exist at one line.
e.g. If I use `|` as row_delimiter.
```
a,b,c,d | e,f
g,h | j,k,l,m
```

Reply:
agree with @ruanwenjun

Reply:
> You cannot deal with rowDelimiter at this way, this can not work in most of case, since not all data will exist at one line. e.g. If I use `|` as row_delimiter.
> 
> ```
> a,b,c,d | e,f
> g,h | j,k,l,m
> ```

I don’t understand why it doesn’t work in most cases. The processing logic is to use custom delimiters to split the read lines. In the example data you gave, the delimiters need to be escaped as “\\| ",There is no problem. Could you please tell me your processing logic?

Reply:
the delimiters need to be escaped as "\\\\|"

Reply:
```
BufferedReader reader =
                new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))
reader.lines()
```
Actually, what worries me more is here. If a 500M file uses `|` as the row separator and there are no `\n` symbols in the file, will the `reader.lines()` method directly read all the contents of the file into memory?

Reply:
If the size of a line in the file is 500M, then the entire line will be read into the memory.  reader.lines() may consume more resources when processing large files, and reader.readLine() should be more suitable.

Reply:
> I don’t understand why it doesn’t work in most cases. The processing logic is to use custom delimiters to split the read lines. In the example data you gave, the delimiters need to be escaped as "\\|",There is no problem. Could you please tell me your processing logic?

For this case, the content will be split into `a,b,c,d`, `e,f`, `g,h`, `j,k,l,m`, am I right?
But he correct content should be `a,b,c,d`, `e,f,g,h`, `j,k,l,m`, you still use `\n` to split the row.

Reply:
> > I don’t understand why it doesn’t work in most cases. The processing logic is to use custom delimiters to split the read lines. In the example data you gave, the delimiters need to be escaped as "\|",There is no problem. Could you please tell me your processing logic?
> 
> For this case, the content will be split into `a,b,c,d`, `e,f`, `g,h`, `j,k,l,m`, am I right? But he correct content should be `a,b,c,d`, `e,f,g,h`, `j,k,l,m`, you still use `\n` to split the row.

you are right. I misunderstood issue.